dart/lib/src/utils/iterable_utils.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:meta/meta.dart';
+
+@internal
+class IterableUtils<T> {
+  IterableUtils(this.iterable);
+
+  Iterable<T>? iterable;
+
+  T? firstWhereOrNull(bool Function(T item) test) {
```

Comment:
Can this be just a static method instead? I guess we don't need to construct a new object every time.

Reply:
I can change it to a static mehthod with generics, but the `T` type of the test functon in not properly resolved, weirdley enough.

<img width="680" alt="Bildschirmfoto 2023-08-14 um 16 00 55" src="https://github.com/getsentry/sentry-dart/assets/3984453/3fb6ade4-9ed1-49b3-86f1-dd6867d694ad">